api/src/main/java/io/minio/MinioClient.java

```
@@ -2591,6 +2598,132 @@ public void deleteObjectTags(DeleteObjectTagsArgs args)
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public ObjectWriteResponse uploadSnowballObjects(UploadSnowballObjectsArgs args)
+      throws ErrorResponseException, InsufficientDataException, InternalException,
+          InvalidKeyException, InvalidResponseException, IOException, NoSuchAlgorithmException,
+          ServerException, XmlParserException {
+    checkArgs(args);
+
+    FileOutputStream fos = null;
+    BufferedOutputStream bos = null;
+    SnappyFramedOutputStream sos = null;
+    ByteArrayOutputStream baos = null;
+    TarArchiveOutputStream tarOutputStream = null;
+
+    try {
+      OutputStream os = null;
+      if (args.stagingFilename() != null) {
+        fos = new FileOutputStream(args.stagingFilename());
+        bos = new BufferedOutputStream(fos);
+        os = bos;
+      } else {
+        baos = new ByteArrayOutputStream();
+        os = baos;
+      }
+
+      if (args.compression()) {
+        sos = new SnappyFramedOutputStream(os);
+        os = sos;
+      }
+
+      tarOutputStream = new TarArchiveOutputStream(os);
+      for (SnowballObject object : args.objects()) {
+        if (object.filename() != null) {
+          Path filePath = Paths.get(object.filename());
+          TarArchiveEntry entry = new TarArchiveEntry(filePath.toFile(), object.name());
+          tarOutputStream.putArchiveEntry(entry);
+          Files.copy(filePath, tarOutputStream);
+        } else {
+          TarArchiveEntry entry = new TarArchiveEntry(object.name());
+          entry.setNames("minio", "minio");
```

This case should return an error instead. Too magic + doesn't work with multiple files anyway.

---

I just added user/group name as `minio`/`minio` for each tar entry in case of stream as source. Do you say this is not needed?

---

Ah. So this is user/group, not object key. No need to add that AFAICT.

---

done